packages/astro/src/core/errors/overlay.ts

```diff
@@ -390,14 +466,31 @@ ${style.trim()}
 </style>
 <div id="backdrop">
   <div id="layout">
+    <div id="theme-toggle-wrapper">
+      <div>
+        <input type="checkbox" class="theme-toggle-checkbox" id="chk" />
+        <label id="theme-toggle-label" for="chk">
+          <svg xmlns="http://www.w3.org/2000/svg" class="icon-tabler icon-tabler-sun" width="15px" height="15px" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
+            <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
+            <circle cx="12" cy="12" r="4" />
+            <path d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7" />
+          </svg>
+          <svg xmlns="http://www.w3.org/2000/svg" class="icon-tabler icon-tabler-moon"  width="15" height="15" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round">
+            <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
+            <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
+          </svg>
+          <div id="theme-toggle-ball"></div>
+        </label>
```

Comment:
And here we add the label to improve the checkbox announcement for screen readers!
```suggestion
          <div id="theme-toggle-ball"></div>
					<span class="sr-only">Use dark theme</span>
        </label>
```